tests/calculations/test_pw.py

```diff
@@ -250,3 +250,26 @@ def test_pw_parallelization_duplicate_cmdline_flag(fixture_sandbox, generate_cal
     with pytest.raises(InputValidationError) as exc:
         generate_calc_job(fixture_sandbox, entry_point_name, inputs)
     assert 'Conflicting' in str(exc.value)
+
+
+def test_environ_namelists(fixture_sandbox, generate_calc_job, generate_inputs_pw):
+    """Test that Environ namelists are  created."""
+    entry_point_name = 'quantumespresso.pw'
+
+    inputs = generate_inputs_pw()
+    inputs['settings'] = orm.Dict(
+        dict={
+            'ENVIRON': {
+                'electrolyte_linearized': True,
+                'environ_type': 'input',
+            },
+            'BOUNDARY': {
+                'solvent_mode': 'electronic',
+                'electrolyte_mode': 'electronic',
+            },
+            'ELECTROSTATIC': {
+                'pbc_correction': 'parabolic',
+            }
+        },
+    )
+    generate_calc_job(fixture_sandbox, entry_point_name, inputs)
```

Comment:
This test is not actually yet testing that the namelists are (properly) generates. The `generate_calc_job` just creates the `CalcJob` and calls the `prepare_for_submission` method. You still need to check that the input file that was generated is correct. To see how to do this, you can look in this same file at the test `test_pw_default` all the way at the top:
```
    with fixture_sandbox.open('aiida.in') as handle:
        input_written = handle.read()
        file_regression.check(input_written, encoding='utf-8', extension='.in')
```
The `file_regression` fixture will compare the input file that was written with a reference file. If that file doesn't yet exist (i.e. before you have ever run this test) it will automatically create it. If you rerun the tests a second time, it will work.

Reply:
Hi @sphuber - thanks a lot for the comments! I have a few questions about how these tests work. I have added in `file_regression` in the most recent commit, but I have not really set up a corresponding input file that it should compare to, and I missing how it automatically creates it. 

Regarding the parser, I modeled the Environ test after the `test_pw_default` test and added in some checks after running an Environ calculation. A very basic question - how does this work in practice, is a calculation actually run - in which case it might break because Environ is a plugin that doesn't ship with the code I believe.

Reply:
If you just add the `file_regression` call in the test, the comparison will be automatically created once you run the tests. If you then run the tests a second time, the comparison file is there and the test should therefore pass. If you ever change anything in the test that would imply that the comparison file has to be updated, you don't have to do this manually either. If you run the tests and they fail because the comparison file is outdated, you can update it by rerunning the tests with the flag `--force-regen` and it will update all comparison files that are used and whose test fails.

> A very basic question - how does this work in practice, is a calculation actually run - in which case it might break because Environ is a plugin that doesn't ship with the code I believe.

No, it doesn't run the calculation. I set up the framework for these parser tests that it simply mocks an already completed calculation with outputs taking from the test directory. Each test has its own folder that corresponds to the name of the test. So all you have to do is add a folder for your test with the output files of a run you did manually with the environ module enabled.

Reply:
Thanks a lot! So I think I understand now how the tests for the calculations work (and have updated the latest commit with the right files). The same thing seems to work for the parser, but I am not sure I am testing this properly. All inputs to Environ would be passed through the settings, so I am not sure if this information is included in the yml that is generated. 
Thanks!

Reply:
The tests now look fine :+1:

Reply:
Great, thanks!